src/tlo/methods/stunting.py

```diff
@@ -275,31 +275,44 @@ def do_recovery(self, idx: Union[list, pd.Index]):
             '-3<=HAZ<-2': 'HAZ>=-2'
         })
 
-    def do_routine_assessment_for_chronic_undernutrition(self, person_id):
-        """This is called by the a generic HSI event for every child aged less than 5 years. It assesses stunting
-        and schedules an HSI as needed."""
-
-        df = self.sim.population.props
-        person = df.loc[person_id]
-        is_stunted = person.un_HAZ_category in ('HAZ<-3', '-3<=HAZ<-2')
-        p_stunting_diagnosed = self.parameters['prob_stunting_diagnosed_at_generic_appt']
-
-        if not is_stunted:
-            return
-
-        # Schedule the HSI for provision of treatment based on the probability of stunting diagnosis
-        if p_stunting_diagnosed > self.rng.random_sample():
-            self.sim.modules['HealthSystem'].schedule_hsi_event(
-                hsi_event=HSI_Stunting_ComplementaryFeeding(module=self, person_id=person_id),
-                priority=2,  # <-- lower priority that for wasting and most other HSI
-                topen=self.sim.date)
-
     def do_treatment(self, person_id, prob_success):
         """Represent the treatment with supplementary feeding. If treatment is successful, effect the recovery
         of the person immediately."""
         if prob_success > self.rng.random_sample():
             self.do_recovery([person_id])
 
+    def do_at_generic_first_appt(
+        self,
+        patient_id: int,
+        patient_details: NamedTuple = None,
+        **kwargs,
+    ) -> IndividualPropertyUpdates:
+        # This is called by the a generic HSI event for every child aged
+        # less than 5 years.
+        # It assesses stunting and schedules an HSI as needed.
+        is_stunted = patient_details.un_HAZ_category in ('HAZ<-3', '-3<=HAZ<-2')
+        p_stunting_diagnosed = self.parameters['prob_stunting_diagnosed_at_generic_appt']
+
+        # Schedule the HSI for provision of treatment based on the
+        # probability of stunting diagnosis, provided the necessary
+        # symptoms are there.
+        if (
+            (patient_details.age_years <= 5)
+            and is_stunted
+            and (p_stunting_diagnosed > self.rng.random_sample())
+        ):
+            # Schedule the HSI for provision of treatment based on the
+            # probability of stunting diagnosis
+            if p_stunting_diagnosed > self.rng.random_sample():
+                event = HSI_Stunting_ComplementaryFeeding(
+                    module=self, person_id=patient_id
+                )
+                self.healthsystem.schedule_hsi_event(
+                    event,
+                    priority=2,  # <-- lower priority that for wasting and most other HSI
+                    topen=self.sim.date,
+                )
```

Comment:
Yep good catch, the Bernoulli RV in the outer `if` statement shouldn't be there, it should only appear in the block below. The original implementation also only generates this RV if `is_stunted` is `True`, which is why there's the seemingly random nested `if` statement (rather than the combined one).